File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dao/BudgetRepository.java
package com.luckyseven.backend.domain.budget.dao;

import com.luckyseven.backend.domain.budget.entity.Budget;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BudgetRepository extends JpaRepository<Budget, Long> {
  Optional<Budget> findByTeamId(Long teamId);
}

File: Backend/src/main/kotlin/com/luckyseven/backend/domain/budget/dao/BudgetRepository.kt
package com.luckyseven.backend.domain.budget.dao

import com.luckyseven.backend.domain.budget.entity.Budget
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.time.LocalDateTime

interface BudgetRepository : JpaRepository<Budget, Long> {
    fun findByTeamId(teamId: Long): Budget?

    @Query("SELECT b.updatedAt FROM Budget b WHERE b.team.id = :teamId")
    fun findUpdatedAtByTeamId(@Param("teamId") teamId: Long): LocalDateTime?
}
